source/server/configuration_impl.h

```diff
@@ -90,7 +91,9 @@ class MainImpl : Logger::Loggable<Logger::Id::config>, public Main {
 
     // Server::Configuration::Listener
     Network::FilterChainFactory& filterChainFactory() override { return *this; }
-    const std::string& address() override { return address_; }
+    Network::Address::InstancePtr address() override {
+      return Network::Utility::resolveUrl(address_);
```

Comment:
Still not right. It doesn't help you if you don't resolve until you access. Resolve at construction time and change address_ to Address::InstancePtr